GraknClient.java

```diff
@@ -1073,4 +1081,72 @@ public int hashCode() {
             return h;
         }
     }
+
+    public enum OptionFlag {
+        INFER,
+        EXPLAIN;
+    }
+
+    private static class OptionsImpl implements OptionsBuilder {
+        private EnumMap<OptionFlag, Boolean> options;
+
+        private OptionsImpl() {
+            options = new EnumMap<>(OptionFlag.class);
+        }
+
+        private OptionsImpl(EnumMap<OptionFlag, Boolean> options) {
+            this.options = options;
+        }
+
+        public OptionsBuilder set(OptionFlag flag, boolean value) {
+            EnumMap<OptionFlag, Boolean> cloned = options.clone();
+            cloned.put(flag, value);
+            return new OptionsImpl(cloned);
+        }
+
+        @Override
+        public OptionsBuilder infer(boolean infer) {
+            return set(OptionFlag.INFER, infer);
+        }
+
+        @Override
+        public OptionsBuilder explain(boolean explain) {
+            return set(OptionFlag.EXPLAIN, explain);
+        }
+
+        @Override
+        public void ifSet(OptionFlag option, FlagConsumer consumer) {
+            Boolean flag = options.get(option);
+            if (flag != null) {
+                consumer.accept(flag);
+            }
+        }
+    }
+
+    public interface OptionsBuilder {
+        OptionsBuilder infer(boolean infer);
+        OptionsBuilder explain(boolean explain);
+
+        OptionsBuilder set(OptionFlag flag, boolean value);
+        void ifSet(OptionFlag option, FlagConsumer consumer);
+
+        @FunctionalInterface
+        interface FlagConsumer {
+            void accept(boolean flag);
+        }
+    }
+
+    public interface Options {
+        OptionsBuilder DEFAULT = new OptionsImpl();
+        OptionsBuilder NO_INFER = DEFAULT.infer(false);
```

Comment:
easier to read without these two extra enums - just use `DEFAULT.explain(...)` and its inverse

Reply:
Removed